Mathlib/LinearAlgebra/CliffordAlgebra/Prod.lean

```diff
@@ -0,0 +1,141 @@
+        induction hm₂ using Submodule.pow_induction_on_left' with
+        | hr =>
+          rw [AlgHom.commutes, Nat.cast_zero, uzpow_zero, one_smul, Algebra.commutes]
+        | hadd _ _ _ _ _ ihx ihy =>
+          rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+        | hmul m₂ hm₂ i x₂ _hx₂ ih₂ =>
+          obtain ⟨v₂, rfl⟩ := hm₂
+          -- this is the second interesting goal
+          rw [map_mul, map_apply_ι, QuadraticForm.Isometry.inr_apply, Nat.cast_succ, ← mul_assoc,
+            ι_mul_ι_comm_of_isOrtho (.inl_inr _ _), neg_mul, mul_assoc, ih₂, mul_smul_comm,
+            ← mul_assoc, ← Units.neg_smul, uzpow_add, uzpow_one, mul_neg_one]
+
+
+/-- The forward direction of `CliffordAlgebra.prodEquiv`. -/
+def ofProd : CliffordAlgebra (Q₁.prod Q₂) →ₐ[R] (evenOdd Q₁ ᵍ⊗[R] evenOdd Q₂) :=
+  lift _ ⟨
+    LinearMap.coprod
+      ((GradedTensorProduct.includeLeft (evenOdd Q₁) (evenOdd Q₂)).toLinearMap
+          ∘ₗ (evenOdd Q₁ 1).subtype ∘ₗ (ι Q₁).codRestrict _ (ι_mem_evenOdd_one Q₁))
+      ((GradedTensorProduct.includeRight (evenOdd Q₁) (evenOdd Q₂)).toLinearMap
+          ∘ₗ (evenOdd Q₂ 1).subtype ∘ₗ (ι Q₂).codRestrict _ (ι_mem_evenOdd_one Q₂)),
+    fun m => by
+      dsimp only [LinearMap.coprod_apply, LinearMap.coe_comp, Function.comp_apply,
+        AlgHom.toLinearMap_apply, QuadraticForm.prod_apply, Submodule.coeSubtype,
+        GradedTensorProduct.includeLeft_apply, GradedTensorProduct.includeRight_apply]
+      simp only [map_add, add_mul, mul_add, GradedTensorProduct.algebraMap_def]
+      rw [GradedTensorProduct.tmul_one_mul_one_tmul, GradedTensorProduct.tmul_one_mul_coe_tmul,
+        GradedTensorProduct.tmul_coe_mul_one_tmul, GradedTensorProduct.tmul_coe_mul_coe_tmul]
+      dsimp
+      simp_rw [one_mul, uzpow_one, Units.neg_smul, one_smul]
+      rw [ι_sq_scalar, ι_sq_scalar, mul_one]
+      simp_rw [← GradedTensorProduct.algebraMap_def, ← GradedTensorProduct.algebraMap_def']
+      abel⟩
+
+@[simp]
+lemma ofProd_ι_mk (m₁ : M₁) (m₂ : M₂) :
+    ofProd Q₁ Q₂ (ι _ (m₁, m₂)) = ι Q₁ m₁ ᵍ⊗ₜ 1 + 1 ᵍ⊗ₜ ι Q₂ m₂ := by
+  rw [ofProd, lift_ι_apply]
+  rfl
+
+/-- The reverse direction of `CliffordAlgebra.prodEquiv`. -/
+def toProd : evenOdd Q₁ ᵍ⊗[R] evenOdd Q₂ →ₐ[R] CliffordAlgebra (Q₁.prod Q₂) :=
+  GradedTensorProduct.lift _ _
+    (CliffordAlgebra.map <| .inl _ _)
+    (CliffordAlgebra.map <| .inr _ _)
+    fun _i₁ _i₂ x₁ x₂ => map_inl_mul_map_inr_of_mem_evenOdd _ _ _ x₁.prop _ x₂.prop
+
+@[simp]
+lemma toProd_ι_tmul_one (m₁ : M₁) : toProd Q₁ Q₂ (ι _ m₁ ᵍ⊗ₜ 1) = ι _ (m₁, 0) := by
+  rw [toProd, GradedTensorProduct.lift_tmul, map_one, mul_one, map_apply_ι,
+    QuadraticForm.Isometry.inl_apply]
+
+@[simp]
+lemma toProd_one_tmul_ι (m₂ : M₂) : toProd Q₁ Q₂ (1 ᵍ⊗ₜ ι _ m₂) = ι _ (0, m₂) := by
+  rw [toProd, GradedTensorProduct.lift_tmul, map_one, one_mul, map_apply_ι,
+    QuadraticForm.Isometry.inr_apply]
+
+lemma toProd_comp_ofProd : (toProd Q₁ Q₂).comp (ofProd Q₁ Q₂) = AlgHom.id _ _ := by
+  ext m₁ <;> dsimp
```

Comment:
Nitpick: we have `m₁ : M₂` in the second goal.
```suggestion
  ext m <;> dsimp
```
Also: shouldn't we squeeze that `dsimp`?